index.js

```diff
@@ -184,6 +184,9 @@ var publicAgent = {
   private_: function() { return agent; }
 };
 
-// TODO: if the module was --require'd on the command line, auto-start the agent
-
 module.exports = publicAgent;
+
+// If the module was --require'd from the command line, start the agent.
+if (module.parent.id === 'internal/preload') {
```

Comment:
Who knows what the future holds, but I think it is okay to use this identifier. If Node makes a breaking change, we'll have to react.